GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -141,6 +143,36 @@ public override void IsServiceInstalledAndRunning(string name, out bool installe
             running = installed && gvfsService.IsRunning;
         }
 
+        public override bool TryCopyPanicLogs(string copyToDir, out string error)
+        {
+            error = null;
+            try
+            {
+                if (!Directory.Exists(DiagnosticReportsDirectory))
+                {
+                    return true;
+                }
+
+                string copyToPanicDir = Path.Combine(copyToDir, ProjFSKext.DriverLogDirectory, "panic_logs");
+                Directory.CreateDirectory(copyToPanicDir);
+
+                foreach (string filePath in Directory.GetFiles(DiagnosticReportsDirectory, PanicFileExtention))
+                {
+                    if (File.ReadAllText(filePath).Contains(ProjFSKext.DriverName))
+                    {
+                        File.Copy(filePath, Path.Combine(copyToPanicDir, Path.GetFileName(filePath)));

```

Comment:
Should we have a try/catch specifically around this call to `File.Copy`?  That way if one copy fails we will can still try to copy any other panics on disk.

Reply:
I'm doing that, but I think that also begs the question should we fail everything if we can't read a log file (surrounding if).  In either scenario I imagine we'll need panics and will end up asking the user to manually copy.

Reply:
I think if copying panic logs fails there's something pretty seriously wrong with the machine, so VFS4G diagnostics are going to be the last of the user's worries.

Reply:
> I'm doing that, but I think that also begs the question should we fail everything if we can't read a log file (surrounding if). In either scenario I imagine we'll need panics and will end up asking the user to manually copy.

I think we should copy what we can but include in the diagnose log if we encountered errors while trying to copy.

Reply:
@wilbaker  @pmj, logging added.  I agree let's make this as easy to diagnose form our side as possible!